### ISSUE TITLE
Unexpected hover result on nested record members

### ISSUE THREAD

**Issue by zth** [jaredly/reason-language-server#419]
Hi!

I'm running into some issues with what's reported in hovers. Here's some example code:

```reason
type owner = {name: string};

type group = {
  id: string,
  name: string,
  owner,
};

type user = {
  id: string,
  memberOf: group,
};

external getTestUser: unit => user = "%identity";
external getTestGroup: unit => group = "%identity";

let make = () => {
  let user = getTestUser();
  let group = getTestGroup();

  user;
  group;
  user.memberOf;
  user.memberOf.owner;
};
```

When I hover `user` and `group` at the bottom of the code block, I get the results I expect in the hover - each type definition is shown as expected. 

When I hover `user` in `user.memberOf`, I also get what I expect - the type definition for user. 

However, when I hover `memberOf` in `user.memberOf`, I *don't* get the definition for `group`, but rather still the definition for `user`. And when I move to the next line and hover `owner` in `user.memberOf.owner`, I get the definition for `group` where I should be getting the definition for `owner`. 

So, to summarize:
`user.memberOf` gives me *user.user* when I should be getting *user.group*
`user.memberOf.owner` gives me *user.user.group* when I should be getting *user.group.owner*

Please let me know if there's anything else I can do to explain this better.
